c_api/src/c_api.rs

```diff
@@ -430,6 +430,33 @@ pub unsafe extern "C" fn c2pa_string_free(s: *mut c_char) {
     }
 }
 
+/// Frees an array of char* pointers created by Rust.
+///
+/// # Parameters
+/// * ptr: pointer to the array of char* pointers.
+/// * count: number of elements in the array.
+///
+/// # Safety
+/// * The ptr passed into this function must point to memory that was allocated
+///   by our library.
+/// * The array and its strings must not have been modified in C.
+/// * The array and its contents can only be freed once and is invalid after this call.
+#[no_mangle]
+pub unsafe extern "C" fn c2pa_free_string_array(ptr: *const *const c_char, count: usize) {
+    if ptr.is_null() {
+        return;
+    }
+
+    let mut_ptr = ptr as *mut *mut c_char;
+    // Free each string directly using the pointer.
+    for i in 0..count {
+        c2pa_string_free(*mut_ptr.add(i));
+    }
+
+    // Free the array.
+    Vec::from_raw_parts(mut_ptr, count, count);
```

Comment:
Is there any way to protect this for bad input?

Reply:
No, I don't think so. We have to manually ensure the invariants of the from_raw_parts are upheld. Namely, that we: 

1. Actually allocated the memory for the vector we are constructing using the Rust allocator
2. No one else is referencing the vector
3. The memory hasn't already been de-allocated.

We check if the ptr is null before we start the de-allocation, and we check if each of the strs are null before de-allocating them.

An additional check we could potentially do is to ensure the length isn't longer than the capacity, but in this case, both the length and capacity are the same.